modules/contact/contact.resolver.js

```diff
@@ -14,23 +17,90 @@ const contactQuery = {
 };
 
 const contactMutation = {
-  addContact: async (parent, args) =>
-    (await contactService.addContact(args.contact)) || {
-      statusCode: 400,
-      message: CONTACT_ALREADY_EXIST,
-    },
+  addContact: async (parent, args) => {
+    try {
+      if (!args.upload) {
+        return await contactService.addContact(args.contact);
+      }
 
-  deleteContact: async (parent, args) =>
-    (await contactService.deleteContact(args.id)) || {
-      statusCode: 404,
-      message: CONTACT_NOT_FOUND,
-    },
+      const uploadResult = await uploadFiles(args.upload);
+      const imageResults = await uploadResult;
 
-  updateContact: async (parent, args) =>
-    (await contactService.updateContact(args.id, args.contact)) || {
-      statusCode: 404,
-      message: CONTACT_NOT_FOUND,
-    },
+      const images = await Promise.all(imageResults).then(res => res);
+      if (!images) {
+        return await contactService.addContact(args.contact);
+      }
+
+      return await contactService.addContact({
+        ...args.contact,
+        images: [
+          { lang: 'uk', value: images[0].fileNames },
+          { lang: 'en', value: images[1].fileNames },
+        ],
+      });
+    } catch (error) {
+      return {
+        statusCode: 400,
+        message: CONTACT_ALREADY_EXIST,
+      };
+    }
+  },
+
+  deleteContact: async (parent, args) => {
+    try {
+      const contact = await Contact.findById(args.id).lean();
+      const deletedImages = await deleteFiles([
+        ...Object.values(contact.images[0].value),
+        ...Object.values(contact.images[1].value),
+      ]);
+
+      if (await Promise.allSettled(deletedImages)) {
+        return await contactService.deleteContact(args.id);
+      }
+    } catch (error) {
+      return {
+        statusCode: 404,
+        message: error.message,
+      };
+    }
+  },
+
+  updateContact: async (parent, args) => {
+    try {
+      if (!args.upload) {
```

Comment:
Please, move all of this logic to the service, let's make resolvers as thin as possible